app/models/task.py

```diff
@@ -1,5 +1,51 @@
 from app import db
 
-
+# many tasks
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable=True)
+    goals = db.relationship("Goal", back_populates="tasks")
+
+
+    def to_dict(self):
+        if self.completed_at is None:
+            if self.goal_id is None:
+                task_dict = {
+                    "id" : self.task_id,
+                    "title" : self.title,
+                    "description" : self.description,
+                    "is_complete" : False
+                }
+            else:
+                task_dict = {
+                    "id" : self.task_id,
+                    "goal_id": self.goal_id,
+                    "title" : self.title,
+                    "description" : self.description,
+                    "is_complete" : False
+                } 
+        else:
+            task_dict = {
+                "id" : self.task_id,
+                "title" : self.title,
+                "description" : self.description,
+                "is_complete" : True
+            }
+        return task_dict
```

Comment:
This is a little extra long.

```suggestion
    def to_dict(self):
        task_dict = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.completed_at == None,
        }
        if self.goal_id is not None:
            task_dict["goal_id"] = self.goal_id
        return task_dict
```